### PR TITLE
Added childclasses and isleaf documentation

### PR DESCRIPTION
**Pull request by dantleech**
Adds documentation for PR https://github.com/doctrine/phpcr-odm/pull/706

### CODE REVIEW

**Comment by dantleech** [comment]
Add docs in the annotation / XML reference

**Comment by dantleech** [comment]
Added reference docs.

File: en/reference/association-mapping.rst [hunk 1]
```diff
@@ -80,6 +80,66 @@ Some sample mappings:
...
+        /** @Document(childClasses={"Article", "Page"}) **/
+        class ContentFolder
+        {
+            // ...
+        }
+
+    .. code-block:: xml
+
+        <doctrine-mapping>
+            <document class="ContentFolder" referenceable="true">
```
Comment (dantleech) [inline en/reference/association-mapping.rst]: Doesn't need to be referencable.

File: en/reference/association-mapping.rst [hunk 2]
```diff
@@ -80,6 +80,66 @@ Some sample mappings:
...
+        /** @Document(isLeaf=true) **/
+        class LeafDocument
+        {
+            // ...
+        }
+
+    .. code-block:: xml
+
+        <doctrine-mapping>
+            <document class="LeafDocument" isLeaf="true">
```
Comment (dantleech) [inline en/reference/association-mapping.rst]: `is-leaf`

File: en/reference/annotations-mapping.rst [hunk 3]
```diff
@@ -95,6 +99,7 @@ Full example:
     *   referenceable=true,
     *   translator="child",
     *   mixins={"mix:created", "mix:lastModified"}
+        childClasses={"SomeDocument", "AnotherDocument"}
```
Comment (dbu) [inline en/reference/annotations-mapping.rst]: lets use App\Documents\Article or something. an example with a namespace, to make it clear that its not relative names or anything.

File: en/reference/association-mapping.rst [hunk 4]
```diff
@@ -80,6 +80,66 @@ Some sample mappings:
...
+
+You may either specify which classes may be children of a document or that a
+document can have no children (i.e. that it be a leaf node).
+
+.. configuration-block::
+
+    .. code-block:: php
+
+        <?php
+        /** @Document(childClasses={"Article", "Page"}) **/
```
Comment (dbu) [inline en/reference/association-mapping.rst]: FQN example with the non-root namespace

**Comment by dbu** [comment]
moved to https://github.com/doctrine/phpcr-odm/pull/781

**Pull request closed without merging**